addons/settings/fnc_init.sqf

```diff
@@ -141,7 +141,7 @@ if (!isNil "_settingInfo") then {
 };
 
 // --- read previous setting values from mission
-_settingsHash = missionNamespace getVariable [QGVAR(hash), "Scenario" get3DENMissionAttribute QGVAR(hash)];
+_settingsHash = getMissionConfigValue QGVAR(hash);
 
 if (isNil "_settingsHash") then {
     _settingsHash = NULL_HASH;
```

Comment:
We could use the alternative syntax of `getMissionConfigValue` to set `NULL_HASH` as default value.

There is one problem with this though. `getMissionConfigValue` only reports something if the mission.sqm exists. This means that all mission settings in 3den use default unless you saved the mission once.

While this makes no difference when previewing the mission (every mission settings use default there currently too), it means that when returning from preview the actual temporary mission settings and what is shown in the settings menu becomes desynched.

test:
`getMissionConfigValue "cba_settings_hash"`
vs.
`"Scenario" get3DENMissionAttribute "cba_settings_hash"`

This could be detected with `missionName` (will report `"tempMissionSP"` and maybe `"tempMissionMP"` in MP 3den).

We could store another copy of these settings in ui namespace if the mission was never saved. idk if that's worth it though.